crates/ra_mbe/src/lib.rs

```diff
@@ -72,10 +110,17 @@ impl MacroRules {
             validate(&rule.lhs)?;
         }
 
-        Ok(MacroRules { rules })
+        Ok(MacroRules { rules, shift: find_subtree_shift(tt, None) })
```

Comment:
this seems like a case where `let shift = find_subtree_shift(tt, None);` and then using field init shorthand would read slightly better.